api/disabled_parameters.go

```diff
@@ -0,0 +1,151 @@
+package api
+
+import (
+	"fmt"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/labstack/echo/v4"
+)
+
+// DisabledParameter holds the name and disabled status of a parameter
+type DisabledParameter struct {
+	Name     string
+	Disabled bool
+}
+
+// DisabledList is a data structure that contains lists of both
+// required and optional parameters.  By storing these, one can
+// determine if any disabled parameters have been supplied
+type DisabledList struct {
+	RequiredParams []DisabledParameter
+	OptionalParams []DisabledParameter
+}
+
+// DisabledMap a type that holds a map of disabled types
+// The key for a disabled map is the handler function name
+type DisabledMap struct {
+	Data map[string]DisabledList
+}
+
+// NewDisabledMap creates a new empty disabled map
+func NewDisabledMap() *DisabledMap {
+	rval := &DisabledMap{}
+	rval.Data = make(map[string]DisabledList)
+	return rval
+}
+
+// NewDisabledMapFromOA3 Creates a new disabled map from an openapi3 definition
+func NewDisabledMapFromOA3(swag *openapi3.Swagger) *DisabledMap {
+	rval := NewDisabledMap()
+	for _, item := range swag.Paths {
+		for _, opItem := range item.Operations() {
+			var disabledList DisabledList
+
+			for _, pref := range opItem.Parameters {
+
+				disabledParameter := DisabledParameter{
+					Name:     pref.Value.Name,
+					Disabled: false,
+				}
+				// TODO how to enable it to be disabled
+
+				if pref.Value.Required {
+					disabledList.RequiredParams = append(disabledList.RequiredParams, disabledParameter)
+					fmt.Printf(" -> Param (Req): %s\n", pref.Value.Name)
```

Comment:
Use the logger for these, they could be verbose messages but probably not necessary for the standard output.

Reply:
Simply removed them...unnecessary really